### DISCUSSION TITLE
Language extensions for js files

### DISCUSSION THREAD

**Discussion by vitor-txt** [CodinGame/monaco-vscode-api#545] [Q&A]
Hello!

Recently, I've noticed that some refactoring options are not shown in my application, as observed in the following images. 

Actual behavior:
![image](https://github.com/user-attachments/assets/7f7d1cf4-9aaf-4d7f-a47d-c4b7d96d2ba7)

Desired behavior:
![image](https://github.com/user-attachments/assets/c7bade09-f28c-4d3b-84af-3c89377755ca)

In my implementation, I use dynamic import/activation of this extensions for js files:
- javascript: @codingame/monaco-vscode-javascript-default-extension,
- typescript-basics: @codingame/monaco-vscode-typescript-basics-default-extension,
- typescript-language-features: @codingame/monaco-vscode-typescript-language-features-default-extension

Do you have any tips on what I can be doing wrong?
Additionally, there's a way to show just one of the libraries (js or ts, preferentially ts) when I hover/click, or can be made just removing one of them?

**Comment by CGNonofr** [comment]
My first guess is it's related to https://github.com/CodinGame/monaco-vscode-api/wiki/Troubleshooting#the-typescript-language-features-extension-is-not-providing-project-wide-intellisense

**Reply by vitor-txt** [reply]
Unfortunately, I've checked this troubleshooting before, and it seems, by the text, that I'm okay with those steps with the vite.config.ts configuration, but the error remains. 
```js
  plugins: [
    {
      // For the *-language-features extensions which use SharedArrayBuffer
      name: 'configure-response-headers',
      apply: 'serve',
      configureServer: (server) => {
        server.middlewares.use((_req, res, next) => {
          res.setHeader('Cross-Origin-Embedder-Policy', 'credentialless');
          res.setHeader('Cross-Origin-Opener-Policy', 'same-origin');
          res.setHeader('Cross-Origin-Resource-Policy', 'cross-origin');
          next();
        });
      },
    },
```

I forgot to add a detail that can be relevant. When I add and activate the typescript-language-features (@codingame/monaco-vscode-typescript-language-features-default-extension) my console shows some messages like this:

```
[vite] Failed to load source map for /home/my_user/project/node_modules/.pnpm/@codingame+monaco-vscode-typescript-language-features-default-extension@11.1.1/node_modules/@codingame/monaco-vscode-typescript-language-features-default-extension/resources/extension.js.
Error: An error occurred while trying to read the map file at https://main.vscode-cdn.net/sourcemaps/undefined/extensions/typescript-language-features/dist/browser/extension.js.map
```

```
Error: ENOENT: no such file or directory, open '/home/my_user/project/node_modules/.pnpm/@codingame+monaco-vscode-typescript-language-features-default-extension@11.1.1/node_modules/@codingame/monaco-vscode-typescript-language-features-default-extension/resources/https:/main.vscode-cdn.net/sourcemaps/undefined/extensions/typescript-language-features/dist/browser/extension.js.map'
    at async open (node:internal/fs/promises:639:25)
    at async Object.readFile (node:internal/fs/promises:1242:14)
    at async extractSourcemapFromFile (file:///home/my_user/path/to/project/node_modules/.pnpm/vite@5.2.13_@types+node@20.14.9_terser@5.31.6/node_modules/vite/dist/node/chunks/dep-DEPSZ3SS.js:16733:10)
    at async loadAndTransform (file://home/my_user/path/to/project/node_modules/.pnpm/vite@5.2.13_@types+node@20.14.9_terser@5.31.6/node_modules/vite/dist/node/chunks/dep-DEPSZ3SS.js:55024:35)
```

```
[vite] Failed to load source map for /home/my_user/path/to/project/node_modules/.pnpm/@codingame+monaco-vscode-typescript-language-features-default-extension@11.1.1/node_modules/@codingame/monaco-vscode-typescript-language-features-default-extension/resources/tsserver.web.js.
Error: An error occurred while trying to read the map file at https://main.vscode-cdn.net/sourcemaps/undefined/extensions/typescript-language-features/dist/browser/tsserver.web.js.map
```

```
Error: ENOENT: no such file or directory, open '/home/my_user/path/to/project/node_modules/.pnpm/@codingame+monaco-vscode-typescript-language-features-default-extension@11.1.1/node_modules/@codingame/monaco-vscode-typescript-language-features-default-extension/resources/https:/main.vscode-cdn.net/sourcemaps/undefined/extensions/typescript-language-features/dist/browser/tsserver.web.js.map'
    at async open (node:internal/fs/promises:639:25)
    at async Object.readFile (node:internal/fs/promises:1242:14)
    at async extractSourcemapFromFile (file:///home/my_user/path/to/project/node_modules/.pnpm/vite@5.2.13_@types+node@20.14.9_terser@5.31.6/node_modules/vite/dist/node/chunks/dep-DEPSZ3SS.js:16733:10)
    at async loadAndTransform (file:///home/my_user/path/to/project/node_modules/.pnpm/vite@5.2.13_@types+node@20.14.9_terser@5.31.6/node_modules/vite/dist/node/chunks/dep-DEPSZ3SS.js:55024:35)

**Reply by CGNonofr** [reply]
if you log `window.crossOriginIsolated` in your app, what does it return?

Those sourcemap errors are unfortunately not suppressable (at least I have no idea how, or we should transform the extensions files to remove the nonexisting sourcemap references)

**Reply by vitor-txt** [reply]
The log of `window.crossOriginIsolated` returns true

**Reply by CGNonofr** [reply]
Ok, are you able to reproduce your issue on the demo here? => https://monaco-vscode-api.netlify.app/?mode=full-workbench

**Reply by vitor-txt** [reply]
No :(

In the demo the refactor works properly with all options I want (the second image showing the desired behavior is from the demo btw).

**Reply by CGNonofr** [reply]
I won't be able to help then without you providing a small reproduction repo :-\

Did you check the VSCode output view?

**Reply by vitor-txt** [reply]
Ok, I'll try to get more details and create a small reproduction to add here. But, thanks a lot for your availability.
